tests/test_plugins.py

```diff
@@ -21,21 +24,50 @@
     "dependencies": [],
 }
 
+TEST_PLUGIN_SCRIPT = """
+import asyncio
+import numpy as np
+from imjoy import api
+
+class ImJoyPlugin():
+    def setup(self):
+        api.log('initialized')
+
+    async def run(self, ctx):
+        await api.alert('Hello')
+        await api.log('done')
+
+    def get_array(self):
+        array = np.zeros(10)
+        # array[3] = 93834.356
+        return {'array': array}
+
+api.export(ImJoyPlugin())
+"""
+
 
 @pytest.fixture(name="test_plugin")
 async def setup_test_plugin(client, event_loop):
     """Initialize the plugin."""
-    pid = await client.init_plugin(TEST_PLUGIN_CONFIG)
-    initialized = event_loop.create_future()
-    client.on_plugin_message(pid, "initialized", initialized)
-    await initialized
-    return {"id": pid}
+    plugin = await client.init_plugin(TEST_PLUGIN_CONFIG)
+    return plugin
 
 
+@pytest.fixture(name="test_plugin_executed")
 async def test_plugin_execute(client, test_plugin, event_loop):
     """Test plugin execute."""
-    pid = test_plugin["id"]
-    executed = event_loop.create_future()
-    await client.execute(pid, {"type": "script", "content": "print('hello')"}, executed)
-    await executed
-    assert executed.result() == {"type": "executeSuccess"}
+    await test_plugin.execute({"type": "script", "content": TEST_PLUGIN_SCRIPT})
+    return test_plugin
+
+
+async def test_plugin_run(client, test_plugin_executed, event_loop):
+    """Test plugin execute."""
+    api = test_plugin_executed.get_api()
+    await api.run({})
```

Comment:
Another strange behavior is that, after running this line, `api.alert("Hello")` in `run` function will be executed 4~5 times.